src/ImageIO/CGImageMetadata.cs

```diff
@@ -119,14 +119,37 @@ public CGImageMetadata (NSData data)
 		}
 
 		[DllImport (Constants.ImageIOLibrary)]
-		extern static void CGImageMetadataEnumerateTagsUsingBlock (/* CGImageMetadataRef __nonnull */ IntPtr metadata,
-			/* CFStringRef __nullable */ IntPtr rootPath, /* CFDictionaryRef __nullable */ IntPtr options,
-			/* __nonnull */ CGImageMetadataTagBlock block);
+		extern unsafe static void CGImageMetadataEnumerateTagsUsingBlock (/* CGImageMetadataRef __nonnull */ IntPtr metadata,
+						/* CFStringRef __nullable */ IntPtr rootPath, /* CFDictionaryRef __nullable */ IntPtr options, BlockLiteral* block);
+
+		delegate bool TrampolineCallback (IntPtr blockPtr, NativeHandle key, NativeHandle value);
+
+		[MonoPInvokeCallback (typeof (TrampolineCallback))]
+		static bool TagEnumerator (IntPtr block, NativeHandle key, NativeHandle value)
+		{
+			var nsKey = Runtime.GetNSObject<NSString> (key, true)!;
+			var nsValue = Runtime.GetINativeObject<CGImageMetadataTag> (value, true)!;
+			var del = BlockLiteral.GetTarget<CGImageMetadataTagBlock> (block);
+			return del (nsKey, nsValue);
+		}
+
+		static unsafe readonly TrampolineCallback static_action = TagEnumerator;
 
 		public void EnumerateTags (NSString? rootPath, CGImageMetadataEnumerateOptions? options, CGImageMetadataTagBlock block)
 		{
 			using var o = options?.ToDictionary ();
-			CGImageMetadataEnumerateTagsUsingBlock (Handle, rootPath.GetHandle (), o.GetHandle (), block);
+			var block_handler = new BlockLiteral ();
+			block_handler.SetupBlockUnsafe (static_action, block);
+
+			unsafe {
+				try {
+					var rootHandle = rootPath is null ? NativeHandle.Zero : rootPath.Handle;
+					var oHandle = o is null ? NativeHandle.Zero : o.GetHandle ();
```

Comment:
This is slightly simpler:

```suggestion
					var rootHandle = rootPath.GetHandle ();
					var oHandle = o.GetHandle ();
```

and yes, GetHandle works on null:

https://github.com/xamarin/xamarin-macios/blob/0517317ae65f39bdb4f16fbf49c573e936061500/src/ObjCRuntime/INativeObject.cs#L25-L28